src/cli/main.rs

```diff
@@ -34,3 +38,22 @@ async fn exec_read_manifest(
     }
     Ok(())
 }
+
+async fn exec_list_manifest(
+    path: &Path,
+    object_store: Arc<dyn ObjectStore>,
+    start: Option<u64>,
+    end: Option<u64>,
+) -> Result<(), Box<dyn Error>> {
+    let range = match (start, end) {
+        (Some(s), Some(e)) => s..e,
+        (Some(s), None) => s..u64::MAX,
```

Comment:
I think you can just do `s..` but I could be wrong. Same for other ones (`..e` and `..`).

Reply:
I wrestled with the type system and even asked GPT because I thought I'd be able to do that, but I get for the former:
```
Type mismatch [E0308] expected `core::ops::range::Range<u64>`, but found `core::ops::range::RangeFrom<u64>` 
```
and the latter:
```
Type mismatch [E0308] expected `core::ops::range::Range<u64>`, but found `core::ops::range::RangeFull` 
```

I tried doing generics the same way it was for the inner method but then I get inference exceptions...

Reply:
I think `R: RangeBounds<u64>` doesn't let you have a union of more than one inner type (e.g. it can be either RangeFull or RangeFrom, but it won't let you instantiate an object that's just that type). 

Obviously I may be missing something, since this seems like it should be possible...

Reply:
Ah, yea, I came across something similar a while back. IIRC there's a way to do it.. there's some base type or something. I can't find it now, though. :/